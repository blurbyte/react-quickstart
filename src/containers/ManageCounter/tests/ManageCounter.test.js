import React from 'react';
import renderer from 'react-test-renderer';

import { ManageCounter, mapDispatchToProps } from '../index';

// mocked components
jest.mock('../CounterPanel', () => 'CounterPanel');

const dispatch = jest.fn();
const actions = mapDispatchToProps(dispatch);

describe('<ManageCounter />', () => {
  it('should render correctly', () => {
    // props init
    const props = {
      counter: 10,
      increaseCounterByAmount: jest.fn(),
      decreaseCounterByAmount: jest.fn()
    };
    // component init
    const component = renderer.create(
      <ManageCounter {...props} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should dispatch increaseCounterByAmount()', () => {
    //init
    const amount = 10;
    actions.increaseCounterByAmount(amount);

    expect(dispatch.mock.calls).toMatchSnapshot();
  });
  it('should dispatch decreaseCounterByAmount()', () => {
    const amount = 0;
    const lowerLimit = 0;
    actions.decreaseCounterByAmount(amount, lowerLimit);

    expect(dispatch.mock.calls).toMatchSnapshot();
  });
  it('should dispatch increaseCounterByAmount() once when handling onIncrease()', () => {
    // props init
    const props = {
      counter: 0,
      increaseCounterByAmount: jest.fn(),
      decreaseCounterByAmount: jest.fn()
    };
    // component init
    const component = renderer.create(
      <ManageCounter {...props} />
    );
    const instance = component.getInstance();
    instance.onIncrease();

    expect(props.increaseCounterByAmount.mock.calls.length).toEqual(1);
  });
  it('should dispatch decreaseCounterByAmount() once when handling onDecrease()', () => {
    // props init
    const props = {
      counter: 20,
      increaseCounterByAmount: jest.fn(),
      decreaseCounterByAmount: jest.fn()
    };
    // component init
    const component = renderer.create(
      <ManageCounter {...props} />
    );
    const instance = component.getInstance();
    instance.onDecrease();

    expect(props.decreaseCounterByAmount.mock.calls.length).toEqual(1);
  });
});
