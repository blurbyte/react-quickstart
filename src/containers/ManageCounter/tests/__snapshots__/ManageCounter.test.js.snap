// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageCounter /> should dispatch decreaseCounterByAmount() 1`] = `
Array [
  Array [
    Object {
      "amount": 10,
      "type": "INCREASE_COUNTER_BY_AMOUNT",
    },
  ],
  Array [
    Object {
      "amount": 0,
      "lowerLimit": 0,
      "type": "DECREASE_COUNTER_BY_AMOUNT",
    },
  ],
]
`;

exports[`<ManageCounter /> should dispatch increaseCounterByAmount() 1`] = `
Array [
  Array [
    Object {
      "amount": 10,
      "type": "INCREASE_COUNTER_BY_AMOUNT",
    },
  ],
]
`;

exports[`<ManageCounter /> should render correctly 1`] = `
<CounterPanel
  counter={10}
  lowerLimit={0}
  onDecrease={[Function]}
  onIncrease={[Function]}
/>
`;
